ion-schema/src/main/kotlin/com/amazon/ionschema/internal/ImportImpl.kt

```diff
@@ -21,14 +21,23 @@ import com.amazon.ionschema.Type
 
 /**
  * Implementation of [Import] for all user-provided ISL.
+ *
+ * This is no longer used to support any of the internal functionality of Ion Schema Kotlin. It exists only for the
+ * [Schema.getImports] and [Schema.getImport] functions.
+ *
+ * Do not call any functions of this class (directly or indirectly) from within the init block of _any_ [Schema]
+ * implementation. If you do, the call may fail, depending on the order that schema imports are resolved, which is not
+ * guaranteed to be stable.
+ *
+ * This class (and [Import]) should be removed in v2.0.0 in favor of a map of schemaId to [Type] (or similar).
```

Comment:
Should the prohibited methods throw? Or can they still be safely called by the user?

Reply:
They can still be called (safely) by the user. The methods will only throw if you call them during the `init { }` block of a `SchemaInternal` implementation.

Reply:
And this particular class is hidden from the user. The user can only see the `Import` interface, so the user won't see this warning.